service/matching/task_queue_partition_manager.go

```diff
@@ -359,6 +359,10 @@ func (pm *taskQueuePartitionManagerImpl) unloadPhysicalQueue(unloadedDbq physica
 		if pm.defaultQueue == unloadedDbq {
 			pm.unloadFromEngine()
 		}
+		// since the entire partition is being removed, the number of
+		// queues it consists of will be the defaultQueue + versionedQueues
+		totalPartitionQueueSize := len(pm.versionedQueues) + 1
+		pm.engine.updatePhysicalTaskQueueGauge(pm, version, -totalPartitionQueueSize)
```

Comment:
this does not pass the right verison. we may have a mixed of build IDs and version sets + the default unversioned one. We should call updatePhysicalTaskQueueGauge one by one for each physical queue instead.
Also all of this should go inside the if block above.
Also you'd want to lock versionedQueuesLock while iterating on it.